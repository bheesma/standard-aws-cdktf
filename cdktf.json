{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "679d25ee-31c0-4b01-9e78-f9e68dcaa0e4",
  "sendCrashReports": "false",
  "terraformProviders": [
    "hashicorp/aws@~> 4.60.0"
  ],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}