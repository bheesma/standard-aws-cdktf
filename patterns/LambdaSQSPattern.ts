import { Construct } from "constructs";
import { StandardLambda } from "../constructs/StandardLambda";
import { StandardSQS } from "../constructs/StandardSQS";


export interface StandardSQSProps {
  queueName: string,
  lambdaName: string
}

export class LambdaSQSPattern extends Construct {

  constructor(scope: Construct, id: string, props: StandardSQSProps) {
    super(scope, id);

    const lambda = new StandardLambda(this, 'stanLambda', {
      lambdaName: props.lambdaName
    });

    const queue = new StandardSQS(this,'stanQueue',{
      queueName: props.queueName
    });

    queue.triggerLambda(lambda);

  }
}
