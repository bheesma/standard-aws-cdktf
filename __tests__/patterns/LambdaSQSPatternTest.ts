// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import "cdktf/lib/testing/adapters/jest"; // Load types for expect matchers
import { Testing } from "cdktf";
import { LambdaSQSPattern } from "../../patterns/LambdaSQSPattern";
import { LambdaEventSourceMapping } from "@cdktf/provider-aws/lib/lambda-event-source-mapping";

describe("LambdaSQSPattern Construct", () => {

  const synthedLambdaSQSPattern = Testing.synthScope((scope) => {
    new LambdaSQSPattern(scope, "app", {
      queueName: "SampleQueue",
      lambdaName: "SampleLambda"
    });
  });


  it("creates lambda with event source mapping", () => {
    expect(synthedLambdaSQSPattern).toHaveResourceWithProperties(LambdaEventSourceMapping, {
      batch_size: 10
      //TODO: Find out how to test Event source mapping ARN
    })
  })

  it("matches the snapshot", () => {
    expect(synthedLambdaSQSPattern).toMatchSnapshot();
  })

});
