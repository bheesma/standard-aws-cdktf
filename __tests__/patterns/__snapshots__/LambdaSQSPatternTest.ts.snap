// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LambdaSQSPattern Construct matches the snapshot 1`] = `
"{
  "data": {
    "aws_iam_policy_document": {
      "app_stanQueue_lambda_sqs_policy_A0E44B47": {
        "statement": [
          {
            "actions": [
              "sqs:SendMessage",
              "sqs:ReceiveMessage",
              "sqs:DeleteMessage",
              "sqs:GetQueueAttributes",
              "sqs:ChangeMessageVisibility"
            ],
            "effect": "Allow",
            "resources": [
              "\${aws_sqs_queue.app_stanQueue_SampleQueue_76D26B69.arn}"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_cloudwatch_log_group": {
      "app_stanLambda_lambda-log-group_4CF5D439": {
        "name": "/aws/lambda/SampleLambda",
        "retention_in_days": 30
      }
    },
    "aws_iam_policy": {
      "app_stanQueue_sqs-policy_F987C306": {
        "name": "LambdaSQSPolicy",
        "policy": "\${data.aws_iam_policy_document.app_stanQueue_lambda_sqs_policy_A0E44B47.json}"
      }
    },
    "aws_iam_role": {
      "app_stanLambda_SampleLambda-role_E5EC063D": {
        "assume_role_policy": "{\\"Version\\":\\"2012-10-17\\",\\"Statement\\":[{\\"Action\\":\\"sts:AssumeRole\\",\\"Principal\\":{\\"Service\\":\\"lambda.amazonaws.com\\"},\\"Effect\\":\\"Allow\\",\\"Sid\\":\\"\\"}]}",
        "name": "SampleLambda-role"
      }
    },
    "aws_iam_role_policy_attachment": {
      "app_stanLambda_lambda-managed-policy_031CB254": {
        "policy_arn": "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
        "role": "\${aws_iam_role.app_stanLambda_SampleLambda-role_E5EC063D.name}"
      },
      "app_stanQueue_execution-role-policy-attachment_88297A78": {
        "depends_on": [
          "aws_iam_policy.app_stanQueue_sqs-policy_F987C306"
        ],
        "policy_arn": "\${aws_iam_policy.app_stanQueue_sqs-policy_F987C306.arn}",
        "role": "\${aws_iam_role.app_stanLambda_SampleLambda-role_E5EC063D.name}"
      }
    },
    "aws_lambda_event_source_mapping": {
      "app_stanQueue_evsourcemap_3A244453": {
        "batch_size": 10,
        "depends_on": [
          "aws_iam_role_policy_attachment.app_stanQueue_execution-role-policy-attachment_88297A78"
        ],
        "event_source_arn": "\${aws_sqs_queue.app_stanQueue_SampleQueue_76D26B69.arn}",
        "function_name": "\${aws_lambda_function.app_stanLambda_SampleLambda_7FC65C63.function_name}"
      }
    },
    "aws_lambda_function": {
      "app_stanLambda_SampleLambda_7FC65C63": {
        "filename": "assets/app_stanLambda_asset_A32CC4EE/A004FBF4B54E411C41147BB16FD0A68B/archive.zip",
        "function_name": "SampleLambda",
        "handler": "lambda_function.lambda_handler",
        "role": "\${aws_iam_role.app_stanLambda_SampleLambda-role_E5EC063D.arn}",
        "runtime": "python3.9",
        "source_code_hash": "A004FBF4B54E411C41147BB16FD0A68B"
      }
    },
    "aws_sqs_queue": {
      "app_stanQueue_SampleQueue-dlq_0F182B15": {
        "name": "SampleQueue-dlq",
        "sqs_managed_sse_enabled": true
      },
      "app_stanQueue_SampleQueue_76D26B69": {
        "name": "SampleQueue",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.app_stanQueue_SampleQueue-dlq_0F182B15.arn}\\",\\"maxReceiveCount\\":5}",
        "sqs_managed_sse_enabled": true
      }
    }
  }
}"
`;
